README.md
# Flutter Color Picker

## Discontinued

- This package has been discontinued in favor of combining all my widgets into a single package for easier maintenance.

- Please see [flutter_widgetz](https://pub.dev/packages/flutter_widgetz) for further updates.


CHANGELOG.md
# Flutter Color Pickr

## 1.0.2

- Discontinued in favor of another package (please see readme).

## 1.0.1

- Fix some lints and add some comments.

## 1.0.0

- Initial release.


test/flutter_color_pickr_test.dart
import 'package:flutter/material.dart';
import 'package:flutter_color_pickr/flutter_color_pickr.dart';
import 'package:flutter_test/flutter_test.dart';

void main() {
  testWidgets('colorPickr', (WidgetTester tester) async {
    await tester.pumpWidget(
      MaterialApp(
        home: Scaffold(
          body: FlutterColorPickr(onTap: (_) {}),
        ),
      ),
    );

    expect(find.text('Choose Color'), findsOneWidget);
  });
}


test/flutter_color_picker_test.dart
import 'package:flutter_test/flutter_test.dart';

import 'package:flutter_color_picker/flutter_color_picker.dart';

void main() {}


lib/flutter_color_picker.dart
import 'package:flutter/material.dart';

class FlutterColorPicker extends StatefulWidget {
  const FlutterColorPicker({
    Key? key,
    required this.initialColor,
    required this.onTap,
  }) : super(key: key);

  final int initialColor;
  final void Function(int) onTap;

  @override
  State<FlutterColorPicker> createState() => _FlutterColorPickerState();
}

class _FlutterColorPickerState extends State<FlutterColorPicker> {
  late int selectedColor;

  static List<Color> get _colors => Colors.primaries.map((MaterialColor color) {
        return color.shade500;
      }).toList();

  @override
  void initState() {
    super.initState();
    selectedColor = widget.initialColor;
  }

  @override
  Widget build(BuildContext context) {
    return Column(
      children: <Widget>[
        const Padding(
          padding: EdgeInsets.all(8.0),
          child: Text('Choose Color'),
        ),
        Expanded(
          child: Container(
            padding: const EdgeInsets.all(20.0),
            child: GridView.builder(
              gridDelegate: const SliverGridDelegateWithMaxCrossAxisExtent(
                maxCrossAxisExtent: 75.0,
                crossAxisSpacing: 10.0,
                mainAxisSpacing: 10.0,
              ),
              itemCount: _colors.length,
              itemBuilder: (BuildContext context, int index) {
                return _Dot(
                  color: _colors[index],
                  selectedColor: Color(selectedColor),
                  onTap: (int value) {
                    setState(() {
                      selectedColor = value;
                    });
                    widget.onTap(value);
                  },
                );
              },
            ),
          ),
        ),
      ],
    );
  }
}

class _Dot extends StatelessWidget {
  const _Dot({
    Key? key,
    required this.color,
    required this.selectedColor,
    required this.onTap,
  }) : super(key: key);

  final Color color;
  final Color selectedColor;
  final void Function(int) onTap;

  @override
  Widget build(BuildContext context) {
    return InkWell(
      onTap: () => onTap(color.value),
      child: Container(
        height: 50.0,
        width: 50.0,
        decoration: BoxDecoration(
          color: color,
          shape: BoxShape.circle,
          border: Border.all(),
        ),
        child: Center(
          child: color == selectedColor
              ? const Icon(
                  Icons.check,
                  color: Colors.white,
                )
              : const SizedBox(),
        ),
      ),
    );
  }
}


lib/src/flutter_color_pickr.dart
import 'package:flutter/material.dart';

class FlutterColorPickr extends StatefulWidget {
  FlutterColorPickr({
    Key? key,
    required this.onTap,
    this.title = 'Choose Color',
    this.initialColor,
    List<Color>? colors,
  }) : super(key: key) {
    this.colors = colors ??
        Colors.primaries.map((MaterialColor color) => color.shade500).toList();
  }

  /// Callback when a color is tapped
  final void Function(Color) onTap;

  /// A title shown above the picker
  final String title;

  /// The initial color for the picker
  final Color? initialColor;

  /// A list of Colors to display
  ///
  /// Defaults to `Colors.primaries`
  late final List<Color> colors;

  @override
  State<FlutterColorPickr> createState() => _FlutterColorPickrState();
}

class _FlutterColorPickrState extends State<FlutterColorPickr> {
  Color? selectedColor;

  @override
  void initState() {
    super.initState();
    selectedColor = widget.initialColor;
  }

  @override
  Widget build(BuildContext context) {
    return Column(
      children: <Widget>[
        Padding(
          padding: const EdgeInsets.all(8.0),
          child: Text(widget.title),
        ),
        Expanded(
          child: Container(
            padding: const EdgeInsets.all(20.0),
            child: GridView.builder(
              gridDelegate: const SliverGridDelegateWithMaxCrossAxisExtent(
                maxCrossAxisExtent: 75.0,
                crossAxisSpacing: 10.0,
                mainAxisSpacing: 10.0,
              ),
              itemCount: widget.colors.length,
              itemBuilder: (BuildContext context, int index) {
                return _Dot(
                  color: widget.colors[index],
                  isSelected: widget.colors[index] == selectedColor,
                  onTap: (Color color) {
                    setState(() {
                      selectedColor = color;
                    });
                    widget.onTap(color);
                  },
                );
              },
            ),
          ),
        ),
      ],
    );
  }
}

class _Dot extends StatelessWidget {
  const _Dot({
    Key? key,
    required this.color,
    required this.isSelected,
    required this.onTap,
  }) : super(key: key);

  final Color color;
  final bool isSelected;
  final void Function(Color) onTap;

  @override
  Widget build(BuildContext context) {
    return InkWell(
      onTap: () => onTap(color),
      child: Container(
        height: 50.0,
        width: 50.0,
        decoration: BoxDecoration(
          color: color,
          shape: BoxShape.circle,
          border: Border.all(),
        ),
        child: Center(
          child: isSelected
              ? const Icon(
                  Icons.check,
                  color: Colors.white,
                )
              : const SizedBox(),
        ),
      ),
    );
  }
}
